Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -0,0 +1,2539 @@
+            let startTimeLabelY = min(thumbStartTimeLabelY, tickLayerStartTimeLabelY)
+            currentExtentStartTimeLabel.frame = CGRect(x: startTimeLabelX, y: startTimeLabelY, width: startTimeLabelSize.width, height: startTimeLabelSize.height)
+        }
+        else {
+            currentExtentStartTimeLabel.string = ""
+            currentExtentStartTimeLabel.isHidden = true
+        }
+        
+        // Update upper label
+        if let endTime = currentExtentEndTime, isRangeEnabled, fullExtent != nil  {
+            let endTimeString = currentExtentDateFormatter.string(from: endTime)
+            currentExtentEndTimeLabel.string = endTimeString
+            let endTimeLabelSize: CGSize = endTimeString.size(withAttributes: [kCTFontAttributeName as NSAttributedStringKey: currentExtentLabelFont])
+            var endTimeLabelX = upperThumbLayer.frame.midX - endTimeLabelSize.width / 2
+            currentExtentEndTimeLabel.isHidden = !isSliderVisible
+            if endTimeLabelX < bounds.origin.x + labelSidePadding && currentExtentStartTime == currentExtentEndTime {
+                endTimeLabelX = bounds.origin.x + labelSidePadding
+                currentExtentEndTimeLabel.isHidden = true
+            }
+            else if endTimeLabelX + endTimeLabelSize.width > bounds.maxX - labelSidePadding {
+                endTimeLabelX = bounds.maxX - endTimeLabelSize.width - labelSidePadding
+                currentExtentEndTimeLabel.isHidden = true
+            }
+            else if endTimeLabelX < currentExtentStartTimeLabel.frame.origin.x + currentExtentStartTimeLabel.frame.width {
+                endTimeLabelX = currentExtentStartTimeLabel.frame.origin.x + currentExtentStartTimeLabel.frame.width + paddingBetweenLabels
+            }
+            
+            let thumbEndTimeLabelY = upperThumbLayer.frame.minY - currentExtentEndTimeLabel.frame.height - labelPadding
+            let tickLayerEndTimeLabelY = tickMarkLayer.frame.minY - currentExtentEndTimeLabel.frame.height - labelPadding
+            let endTimeLabelY = min(thumbEndTimeLabelY, tickLayerEndTimeLabelY)
+            currentExtentEndTimeLabel.frame = CGRect(x: endTimeLabelX, y: endTimeLabelY, width: endTimeLabelSize.width, height: endTimeLabelSize.height)
+        }
+        else {
+            currentExtentEndTimeLabel.string = ""
+            currentExtentEndTimeLabel.isHidden = true
+        }
+        
+        // Commit the transaction
+        CATransaction.commit()
+    }
+    
+    private func positionTickMarks() {
+        //
+        // Bail out if time steps are not available
+        guard let timeSteps = timeSteps, timeSteps.count > 0, isSliderVisible else {
+            return
+        }
+        
+        // Create tick marks based on
+        // time steps and it's calculated
+        // position on the slider
+        var tms = [TickMark]()
+        for i in 0..<timeSteps.count  {
```

Comment:
While this `for-in` loop works, it could be improved upon. It assumes that the `startIndex` of `timeSteps` is `0`. That is true for `Array`, but not true for all types that conform to `Collection`. Rather than constructing the index range manually, it is better to use the `indices` property of `Collection`: `for i in timeSteps.indices { … }`. And if you don't actually need the index, which is the case here, you could just use `for timeStep in timeSteps { … }`.

Reply:
fixed